utils/watchcat/files/watchcat.sh

```diff
@@ -54,6 +54,18 @@ get_ping_family_flag() {
 	echo $family
 }
 
+get_ping_timeout() {
+	timeout=$1
+	case "$timeout" in
+	''|*[!0-9]*)
+		# Fallback to default value if not an integer
+		timeout=10
+		;;
+	esac
+
+	echo $timeout
+}
+
```

Comment:
You can avoid this altogether by supplying a default if timeout is empty or unsupplied. 

`${parameter:-[word]}`. Param is empty? Fill with default `[word]`

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html

Reply:
This is what I've done first but then changed my mind. The code here also defaults to the default in case someone pass it a non-integer or negative value. If we don't care about this case then yes, shell default value syntax will be simpler 🙂. WDYT?

Reply:
@systemcrash  I pushed a new commit to change that part. But honestly I'm not convinced.

That will silently fail with such case
```
# use of "A" / "-1" / ""1.5" as last param is not a positive integer and will silently fail the test
sh -x /usr/bin/watchcat.sh restart_iface 30 1.1.5.5 3 small wwan0 free 0 ipv4 A

# would translate to 
ping -4 -I wwan0 -s 1 -c 1 -W A 1.1.5.5

# previous implementation would have set the default value
ping -4 -I wwan0 -s 1 -c 10 -W 10 1.1.5.5
```

Reply:
thanks @Neustradamus 

@systemcrash @Neustradamus Regarding comments above. We still need to decide on ab35bd78713a838fcc0aa3b877ac0d1140a62b73 (xref https://github.com/openwrt/packages/pull/25174#discussion_r1809225113). I would prefer to remove it and keep previous implementation. But final word is yours :)

Then we have https://github.com/openwrt/luci/pull/7337 to check. (It's now having merge conflicts but I will resolve them lately)

Thanks